File: t1_service_with_notifications/src/main/java/lord/main/t1_service_with_notifications/supporting_classes/RecyclerViewItem.java
package lord.main.t1_service_with_notifications.supporting_classes;

import lord.main.t1_service_with_notifications.ActivityShowNotifyHistory;

/**
 * Шаблон со всеми необходимыми данными для элемента RecyclerView в {@link ActivityShowNotifyHistory активности демонстрации истории уведомлений}
 */
public class RecyclerViewItem {
    /**
     * Уникальный идентификатор уведомления
     */
    public int id,
    /**
     * Уникальный идентификатор иконки из ресурсов
     */
    iconId,
    /**
     * Цвет заливки иконки
     */
    iconColor;
    /**
     * Уникальный идентификатор канала уведомлений
     */
    public String channelId,
    /**
     * Заголовок уведомления
     */
    title,
    /**
     * Текст уведомления
     */
    text;
}

File: t1_service_with_notifications/src/main/java/lord/main/t1_service_with_notifications/supporting_classes/RecyclerViewItem.kt
package lord.main.t1_service_with_notifications.supporting_classes

import lord.main.t1_service_with_notifications.ActivityShowNotifyHistory

/**
 * Шаблон со всеми необходимыми данными для элемента RecyclerView в [активности демонстрации истории уведомлений][ActivityShowNotifyHistory]
 */
object RecyclerViewItem {
    /**
     * Уникальный идентификатор уведомления
     */
    var id = 0

    /**
     * Уникальный идентификатор иконки из ресурсов
     */
    var iconId = 0

    /**
     * Цвет заливки иконки
     */
    var iconColor = 0

    /**
     * Уникальный идентификатор канала уведомлений
     */
    var channelId = ""

    /**
     * Заголовок уведомления
     */
    var title = ""

    /**
     * Текст уведомления
     */
    var text = ""
}